File: frontend/FamoNET.Scripts/Model/ChartParameters.ts
import { AxisMode } from "./Enums.js";
import { ViewportParameters } from "./ViewportParameters.js"

export class ChartParameters {    
    Title: string = "";
    Logarithmic: boolean = false;    
    DisableXLabels: boolean = false;
    DisableEvents: boolean = false;
    InvertYAxis: boolean = false;
    AxisMode: AxisMode = AxisMode.MJD

    public constructor(title: string, logarithmic?: boolean, disableXLabels?: boolean, disableEvents?: boolean, invertYAxis?: boolean, axisMode?: AxisMode) {        
        this.Title = title;
        if (disableXLabels) {
            this.DisableXLabels = disableXLabels;
        }

        if (logarithmic) {
            this.Logarithmic = logarithmic;       
        }        

        if (disableEvents) {
            this.DisableEvents = disableEvents;
        }

        if (invertYAxis) {
            this.InvertYAxis = invertYAxis;
        }

        if (axisMode) {
            this.AxisMode = axisMode;
        }
    }
}

File: frontend/FamoNET/Model/ChartParameters.cs
namespace FamoNET.Model
{
    public class ChartParameters<T>
    {
        public string Title { get; set; }
        public ViewportParams<T> Viewport { get; set; } = new ViewportParams<T>();
        public double Offset { get; set; }        
        public bool Logarithmic { get; set; }
        public bool DisableXLabels {  get; set; }
        public bool DisableEvents { get; set; }
        public bool InvertYAxis { get; set; }
        public AxisMode AxisMode { get; set; }
        public ChartParameters()
        {
            
        }

        public ChartParameters(ChartParameters<T> chartParams)
        {
            Title = chartParams.Title;
            Viewport = new ViewportParams<T>(chartParams.Viewport);
            Offset = chartParams.Offset;            
            Logarithmic = chartParams.Logarithmic;
            DisableEvents = chartParams.DisableEvents;
            InvertYAxis = chartParams.InvertYAxis;
            AxisMode = chartParams.AxisMode;
        }
    }
}
